eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/BbodySpec.hs

```diff
@@ -102,25 +94,149 @@ spec = describe "BBODY" $ do
                   }
             )
         ]
+
+  it "BodyRefScriptsSizeTooBig with reference scripts in the same block" $
+    whenMajorVersionAtLeast @12 $ do
+      Just plutusScript <- pure $ mkPlutusScript @era $ purposeIsWellformedNoDatum SPlutusV2
+      let scriptSize = originalBytesSize plutusScript
+
+      txScriptCounts <-
+        genNumAdditionsExceeding
+          scriptSize
+          maxRefScriptSizePerTx
+          maxRefScriptSizePerBlock
+
+      let expectedTotalRefScriptSize = scriptSize * sum txScriptCounts
+
+      -- We are creating reference scripts and transaction that depend on them in a "simulation",
+      -- so the result will be correctly constructed that are not applied to the ledger state
+      txs :: [Tx era] <- simulateThenRestore $ do
+        concat
+          <$> forM
+            txScriptCounts
+            ( \n -> do
+                -- produce reference scripts
+                refScriptTxs <-
+                  replicateM n (produceRefScriptsTx (fromPlutusScript plutusScript :| []))
+
+                -- spend using the reference scripts
+                let txIns = (`mkTxInPartial` 0) . txIdTx <$> refScriptTxs
+                rootIn <- fst <$> getImpRootTxOut
+                spendTx <- spendUsingRefInputs rootIn (NE.fromList txIns)
+                pure $ refScriptTxs ++ [spendTx]
+            )
+
+      Left predFailures <- tryRunBBODY txs
```

Comment:
I'd suggest using `expectLeft` or `expectLeftExpr`, since they produce a more sensible test failure, instead of generic MonadFail failure:
```suggestion
      predFailures <- expectLeftExpr =<< tryRunBBODY txs
```

Reply:
Nice! Done, thank you